lib/fork.py

```diff
@@ -84,24 +90,30 @@ def add(i, j): return i+j
     return {"r": r, "pid": pid}
 
 
-###############################
 def fetch_fork_result(r, pid):
-    """Used with fork clients to retrieve results
-    Can raise:
-    OSError if Bad file descriptor or file already closed or if waitpid syscall returns -1
-    FetchError if a os.read error was encountered
-    Possible errors from os.read and pickle.load (catched here):
-    - EOFError if the forked process failed an nothing was written to the pipe, if cPickle finds an empty string
-    - IOError failure for an I/O-related reason, e.g., "pipe file not found" or "disk full"
-    - OSError other system-related error (includes both former OSError and IOError since Py3.4)
-    - pickle.UnpicklingError incomplete pickled data
+    """Used with fork clients to retrieve results.
+
+    Can raise OSError and FetchError.
+    Other errors can come from `os.read` and `pickle.load` (but are caught here):
+      - EOFError if the forked process failed an nothing was written to the pipe, if cPickle finds an empty string
+      - IOError failure for an I/O-related reason, e.g., "pipe file not found" or "disk full"
+      - OSError other system-related error (includes both former OSError and IOError since Py3.4)
+      - pickle.UnpicklingError incomplete pickled data
 
     Args:
-        r (pipe): Input pipe
-        pid (int): pid of the child
+        r (int): Input pipe.
+        pid (int): PID of the child.
 
     Returns:
-        Object: Unpickled object
+        object: Unpickled object.
+
+    Raises:
+        FetchError: If an os.read error was encountered or if the forked process failed.
+        OSError: If Bad file descriptor or file already closed or if waitpid syscall returns -1
+        EOFError: If the forked process failed an nothing was written to the pipe, if cPickle finds an empty string
+        IOError: Failure for I/O-related reason, E.g., "pipe file not found" or "disk full"
+        OSError: Other system-related error (includes both former OSError and IOError since Py3.4)
+        pickle.UnpicklingError: Incomplete pickled data
```

Comment:
Should all these items end with a period `.` (the first one does)?

Reply:
done